### ISSUE TITLE
Add ability to create or update application password credential

### ISSUE THREAD

**Issue by wouterroos** [LockTar/AzureAdApplicationManagement#15]
Hi,

Currently, there is no way to generate secrets using the steps provided. A colleague of mine has already done some work to figure out how this could be done using PowerShell (see attachment) but we're having an issues with using `Connect-AzureAD` in Azure DevOps, which your extension has already solved. 

Would it be possible to incorporate this functionality?

Thanks in advance,

Regards,
Wouter

[CreateOrReplace-AzureAdApplicationPasswordCredential.txt](https://github.com/LockTar/AzureAdApplicationManagement/files/2514584/CreateOrReplace-AzureAdApplicationPasswordCredential.txt)

**Comment by wouterroos** [comment]
@LockTar Hi, It's been a while but I again find myself in need of this functionality and I'm willing to assist in implementing it. Do you have an idea on how to implement this feature? As it's possible to have multiple secrets in an Application, I would suggest adding the possibility to pass one or more secret 'Descriptions' (see image) after which the tool would generate a secret for each description, overwriting any existing keys with an existing description. Thoughts?

![image](https://user-images.githubusercontent.com/5137829/56812401-18cda400-683b-11e9-9f10-19f7865b9a6e.png)

**Comment by LockTar** [comment]
I've been thinking about this as well. I think we have multiple solutions. 
Possible solutions and give feedback as you want:

1. A "SetKey" task that has as input a description and a datetime for as enddate. This task will always overwrite the key with that description (case sensitive) and will create one if it doesn't exist. If multiple keys found with the same name, remove all and create new key (overwrite all keys to 1 new key).
We can create some kind of a list (not really easy because dynamic list objects aren't documented) but we can hardcode a list of 5 or 10 keys. Unlikely we get demand for more than 10. And if so, easy to extend.
2. A "CreateKey" task that will only create a key with enddate and description. Sometimes you just need to create a key once and pass it to something else. You could run this task only once with a condition. The only thing is, what is that condition. Well, we could extend the "SetApplication" and "CreatApplication" task that we already have. That Set task checks if an application exist, if not, it will create it. At that moment we could set an output variable "IsNowCreated" (do you have better name) to true. You could use that in the condition. 

We can also create both tasks. Not much work to add both because logic is almost the same.

In both tasks I would also add a output variable that will return of course the generated key/secret.

**Comment by wouterroos** [comment]
If I look at my own use-case, adding this functionality to the Set/CreateApplication task would make the most sense. It would then look like just another 'property' on said tasks. As for the input format, we could perhaps use a simple JSON object to represent the secrets name and enddate, this way we're not hard-coding anything and we can pass all secret information as a single text input. I have time today to get this to work so I'll fork your repo and start laying the groundwork for a pull request.

**Comment by wouterroos** [comment]
@LockTar see #30

**Comment by LockTar** [comment]
@wouterroos I changed the set variable so it is a secret. Otherwise it was showing up in logfiles etc. Can you test if this is working for you?

**Comment by wouterroos** [comment]
@LockTar I just ran the new step and this this causes an issue for my pipeline as downstream steps consume the secret as an environment variable, which doesn't work if it is a secret variable. I do however get your concern that you might not want to expose the keys in logs etc.

**Comment by wouterroos** [comment]
> @LockTar I just ran the new step and this this causes an issue for my pipeline as downstream steps consume the secret as an environment variable, which doesn't work if it is a secret variable. I do however get your concern that you might not want to expose the keys in logs etc.
> 
> I'm currently looking for a solution for my issue so I'll keep you posted.

@LockTar I was able to work around this specific issue using the method explained [here](https://stackoverflow.com/a/50374601/1662317) so I'm fine with setting it as a secret variable in the step!
